src/realm/object-store/sync/sync_session.cpp

```diff
@@ -232,73 +234,124 @@ SyncSession::handle_refresh(const std::shared_ptr<SyncSession>& session)
 SyncSession::SyncSession(SyncClient& client, std::shared_ptr<DB> db, SyncConfig config, SyncManager* sync_manager)
     : m_config(std::move(config))
     , m_db(std::move(db))
-    , m_flx_subscription_store(make_flx_subscription_store())
+    , m_flx_subscription_store([this](bool use_flx_sync) -> std::shared_ptr<sync::SubscriptionStore> {
```

Comment:
Hoisting the load of `m_config.flx_sync_requested` out of the function is the functional change here. I initially thought that moving this to an immediately-invoked lambda was required, but it actually isn't, and could be reverted if desired.